src/components/dashboard/send-message.svelte

```diff
@@ -0,0 +1,58 @@
+<script>
+  import { getContext } from 'svelte';
+  import { TextField, Button } from 'attractions';
+  import { snackbarContextKey } from 'attractions/src/snackbar';
+  import * as api from '@/utils/api.js';
+
+  const showSnackbar = getContext(snackbarContextKey);
+
+  export let users = [];
+  export let account;
+
+  let text = null;
+
+  async function send() {
+    if (!text) {
+      return;
+    }
+
+    try {
+      const promises = users.map(user => api.json(api.post(`/accounts/${user.email}/notify`, {
+        data: {
+          message: text,
+        },
+        csrfToken: account.csrf_token,
+      })));
+      await Promise.all(promises);
+      showSnackbar({ props: { text: `Successfully sent the message.` } });
+      text = '';
+    } catch (e) {
+      console.error(e);
+      showSnackbar({ props: { text: 'Something went wrong, try reloading the page.' } });
+    }
+  }
+</script>
+
+<div class="send-message">
+  <header>
+    <svg src="images/icons/message-circle.svg" class="icon mr-2" />
+    Send a message
+  </header>
+  <form>
+    <TextField
+      bind:value={text}
+      multiline
+      maxlength={512}
+      cols="5"
+      id="send-message"
```

Comment:
where is this id being used? and wouldn't it cause confusion that it is the same as the class assigned to the root element?

Reply:
It was just screaming at me for having duplicate IDs `null` for some reason

Reply:
It seems like an SSR issue. This only happens in Sapper, and gets immediately fixed up on hydration. It's still a nasty warning in the console, so I'd like to avoid it if possible, though I must agree that this is simply treating the symptoms

Reply:
And it is only happening with Attractions, so I'll investigate that more. Probably a bug on our side

Reply:
That is because of the default strigification behavior of Javascript, and the prop's [default value is `null`](https://github.com/illright/attractions/blob/develop/src/text-field/text-field.svelte#L16). This should be fixed from the component itself, and I think `id` should be part of `$$restProps`, with the label surrounding the input.
If you were thinking about empty string as default, [it is not allowed](https://html.spec.whatwg.org/multipage/dom.html#the-id-attribute)

Reply:
No, Svelte specifically doesn't set attributes if their values are nullish